### PR TITLE
Manticore 0.2.3

### PR DESCRIPTION
**Pull request by mossberg**
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/1298)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,16 +2,45 @@
...
+- Ethereum: `--txpreconstrain` CLI flag. Enabling this avoids sending gas to nonpayable functions, primarily avoiding exploration of uninteresting revert states.
+- Research memory model (LazySMemory) allowing for symbolic memory indexing to be handled without concretization (opt in, currently for research only)
 
 ### Changed
 
 - Linux/binary analysis has been moved to `manticore.native`, `manticore.core.cpu` has been moved to `manticore.native.cpu`. Please update your imports.
-- The binary analysis dependencies are now not installed by default. They can be installed with `pip install manticore[native]`.
+- The binary analysis dependencies are now not installed by default. They can be installed with `pip install manticore[native]`. This is to prevent EVM users from installing binary dependencies.
 - The symbolic `stdin_size` is now a config variable (in `main` config group) with a default of 256 (it was like this before).
+- ManticoreEVM.generate_testcase ‘name’ parameter is now optional
```
Comment (yan) [inline CHANGELOG.md]: add `()`?

File: README.md [hunk 2]
```diff
@@ -12,14 +12,14 @@
 [![Maintainability](https://api.codeclimate.com/v1/badges/9161568d8378cea903f4/maintainability)](https://codeclimate.com/github/trailofbits/manticore/maintainability)
 [![Test Coverage](https://api.codeclimate.com/v1/badges/9161568d8378cea903f4/test_coverage)](https://codeclimate.com/github/trailofbits/manticore/test_coverage)
 
-Manticore is a symbolic execution tool for analysis of binaries and smart contracts.
+Manticore is a symbolic execution tool for analysis of smart contracts and binaries.
```
Comment (yan) [inline README.md]: Since we now more fully support macOS, do we want to specify "Linux binaries"?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,16 +2,45 @@
...
-- The binary analysis dependencies are now not installed by default. They can be installed with `pip install manticore[native]`.
+- The binary analysis dependencies are now not installed by default. They can be installed with `pip install manticore[native]`. This is to prevent EVM users from installing binary dependencies.
 - The symbolic `stdin_size` is now a config variable (in `main` config group) with a default of 256 (it was like this before).
+- `ManticoreEVM.generate_testcase()` 'name' parameter is now optional
+- Manticore CLI run on a smart contract will now use all detectors by default (detectors can be listed with --list-detectors, excluded with --exclude <detectors> or --exclude-all)
+- Misusing the ManticoreEVM API, for example by using old keyword arguments that are not available since some versions (like ManticoreEVM(verbosity=5)) will now raise an exception instead of not applying the argument at all.
 
+### Fixed
+
+- Fixed timeout support for ManticoreEVM
```
Comment (yan) [inline CHANGELOG.md]: `ManticoreEVM`/`Manticore` difference is pretty low level for a changelog; just leave as 'Fixed timeout handling'?

File: README.md (hunk 2 above)
Comment (mossberg) [reply README.md]: i'd prefer to keep it as is, i think that makes this line look a little cluttered, and i think it's ok without

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,16 +2,45 @@
...
+- [PetarMI](https://github.com/PetarMI)
+- [hidde-jan](https://github.com/hidde-jan)
+- [catenacyber](https://github.com/catenacyber)
+
+### Added
+
+- Support for ARM THUMB instructions: ADR, ADDW, SUBW, CBZ, TBB, TBH, STMDA, STMDB
+- `State.solve_minmax()` API for querying a BitVec for its min/max values
+- New SMTLIB optimization for simplifying redundant concat/extract combinations; helps reduce expression complexity, and speed up queries
+- Ethereum: `--txpreconstrain` CLI flag. Enabling this avoids sending gas to nonpayable functions, primarily avoiding exploration of uninteresting revert states.
```
Comment (feliam) [inline CHANGELOG.md]: sending ETHER/value not gas.

File: CHANGELOG.md (hunk 3 above)
Comment (mossberg) [reply CHANGELOG.md]: it's specifically for the eth side, i can do something like 

```suggestion
- Ethereum: Fixed CLI timeout support
```

File: CHANGELOG.md (hunk 4 above)
Comment (mossberg) [reply CHANGELOG.md]: great catch

**Pull request merged**